### PR TITLE
A CLI shell for adhoc queries.

### PR DESCRIPTION
**Pull request by gz**
And a few CLI improvements.

### CODE REVIEW

File: crates/fda/src/main.rs [hunk 1]
```diff
@@ -132,51 +132,50 @@ async fn api_key_commands(action: ApiKeyActions, client: Client) {
...
+                .send()
+                .await
+                .map_err(handle_errors_fatal("Failed to list API keys", 1))
+                .unwrap();
+            let mut rows = vec![];
+            rows.push(["name".to_string(), "id".to_string()]);
+            for key in response.iter() {
+                rows.push([key.name.to_string(), key.id.0.to_string()]);
+            }
+            println!(
```
[the remark this answers is not in the document]
Comment (gz) [reply crates/fda/src/main.rs:152]: Ill add support for both json and text output over the next few iterations.

File: crates/fda/src/shell.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    if let Some(config_dir) = config_dir {
+        let _r = rl.load_history(&config_dir.join(HISTORY_FILE));
+    }
+
+    let prompt = format!("{}> ", name);
+    loop {
+        let readline = rl.readline(prompt.as_str());
+        match readline {
+            Ok(line) => {
+                let trimmed_line = line.trim();
```
[the remark this answers is not in the document]
Comment (gz) [reply crates/fda/src/shell.rs:53]: can you give me an example if what you would like to do?

File: crates/fda/src/shell.rs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                break;
+            }
+            Err(err) => {
+                println!("ERROR: {:?}", err);
+                break;
+            }
+        }
+    }
+
+    if let Some(config_dir) = config_dir {
```
[the remark this answers is not in the document]
Comment (gz) [reply crates/fda/src/shell.rs:112]: No ! command yet, just arrow up to cycle through history. Adding failed commands too is standard for shells (bash, fish, zsh all do it) makes it easy to resend+fix the prev command if it was wrong

File: crates/fda/src/shell.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mihaibudiu) [reply crates/fda/src/shell.rs:53]: Commands longer than 1 line

[the remark this answers is not in the document]
Comment (gz) [reply crates/fda/src/shell.rs:53]: You can paste multiple lines with ctrl+v, for more multi-line support I'll need to look how e.g., the UX is in psql.

**Pull request merged**